### PR TITLE
Create finishReleaseCycle workflow

### PR DESCRIPTION
**Pull request by roryabraham**
### Details
This is _slightly_ different from the design doc in that we're going to be deploying to staging/production only when the corresponding branch has actually been updated. So we have two workflows, `finishReleaseCycle` and `deploy`, rather than just the one `productionDeployCash` outlined in the doc. The high-level process is the same:

1) QA is completed, so the `StagingDeployCash` is closed
2) That triggers the `finishReleaseCycle` workflow, which:
    i) Creates an automerge pull request from staging -> production
    ii) Generates a new BUILD version on master and deploys staging (i.e: deploy all the PRs that were deferred during QA)
    iii) Creates a new `StagingDeployCash`
3) When the automerge PR from (2i) is merged and the production branch is actually updated, then the `deploy` workflow is triggered and we'll create a release to perform the actual production deploy.

### Fixed Issues
Fixes (partial) https://github.com/Expensify/Expensify/issues/155221

### Tests
To test this workflow, we'll need to:

1) Merge this PR
2) Close an E.cash issue labeled `StagingDeployCash`

We'll then expect that:

- [ ] An automerge PR is created from staging -> production
- [ ] A new `BUILD` version is created off master.
- [ ] An automerge PR for that new version is created w/ master as the base. Only the version files should be updated in that PR, and it should be merged.
- [ ] Tags should be pushed, and a staging deploy should happen
- [ ] A new StagingDeployCash should be triggered.

### CODE REVIEW

File: .github/workflows/finishReleaseCycle.yml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.OS_BOTIFY_TOKEN }}
+
+      - name: Set up git config
+        run: git config user.name OSBotify
+
+      - name: Checkout staging branch
+        run: git checkout staging
```
Comment (AndrewGable) [inline .github/workflows/finishReleaseCycle.yml:26]: I think you can do this in `actions/checkout`

https://github.com/actions/checkout#Checkout-a-different-branch

Comment (roryabraham) [reply .github/workflows/finishReleaseCycle.yml:26]: Do you know if you can provide a `ref` and the `fetch-depth: 0` to checkout a particular branch but also fetch the full branch history?

Comment (AndrewGable) [reply .github/workflows/finishReleaseCycle.yml]: I don't see anywhere saying you explicitly can't, so I would assume you can

**Comment by roryabraham** [comment]
Updated!

**Review by AndrewGable — APPROVED** [review]

**Comment by botify** [comment]
@AndrewGable looks like this was merged without passing tests. Please add a note explaining why this was done or remove the `Emergency` label if this is not an emergency.

**Comment by AndrewGable** [comment]
Merged to continue getting deploys moving and grooving 🕺

**Comment by roryabraham** [comment]
Workflow was triggered when I closed the `StagingDeployCash`: https://github.com/Expensify/Expensify.cash/runs/2143280721?check_suite_focus=true

**Comment by roryabraham** [comment]
First run failed because the `production` branch wasn't pushed yet.

[Second run] created the [automerge PR](https://github.com/Expensify/Expensify.cash/pull/1865) (bad title), but that failed to automerge, even though the automerge workflow reported no failure.

**Comment by roryabraham** [comment]
Retesting this after fixing some bugs:

finishReleaseCycle workflow: https://github.com/Expensify/Expensify.cash/runs/2144100874?check_suite_focus=true ❌  failed

**Comment by roryabraham** [comment]
This failed because the `bumpVersion` action won't work on ubuntu 🤦🏼‍♂️

**Comment by roryabraham** [comment]
Continuing testing in [this PR](https://github.com/Expensify/Expensify.cash/pull/1876)

**Pull request merged**